smiles/smiles.g4

```diff
@@ -1,561 +1,369 @@
-   | (UH LO)
-   | (UE LR)
-   | (UT LM)
-   | (UY LB)
-   | (UL LU)
-   | (UA LC)
-   | (UT LH)
-   | (UP LA)
-   | (UU)
-   | (UN LP)
-   | (UP LU)
-   | (UA LM)
-   | (UC LM)
-   | (UB LK)
-   | (UC LF)
-   | (UE LS)
-   | (UF LM)
-   | (UM LD)
-   | (UN LO)
-   | (UL NR)
-   ;
-
-aromatic_symbol
-   : LC
-   | LN
-   | LO
-   | LP
-   | LS
-   | (LS LE)
-   | (LA LS)
-   ;
-
-chiral
-   : '@'
-   | '@@'
-   | '@TH1'
-   | '@TH2'
-   | '@AL1'
-   | '@AL2'
-   | '@SP1'
-   | '@SP2'
-   | '@SP3'
-   | '@TB1'
-   | '@TB2'
-   | '@TB3'
-   | '@TB3'
-   | '@TB4'
-   | '@TB5'
-   | '@TB6'
-   | '@TB7'
-   | '@TB8'
-   | '@TB9'
-   | '@TB10'
-   | '@TB11'
-   | '@TB12'
-   | '@TB13'
-   | '@TB14'
-   | '@TB15'
-   | '@TB16'
-   | '@TB17'
-   | '@TB18'
-   | '@TB19'
-   | '@TB20'
-   | '@TB21'
-   | '@TB22'
-   | '@TB23'
-   | '@TB24'
-   | '@TB25'
-   | '@TB26'
-   | '@TB27'
-   | '@TB28'
-   | '@TB29'
-   | '@TB30'
-   | '@OH1'
-   | '@OH2'
-   | '@OH3'
-   | '@OH4'
-   | '@OH5'
-   | '@OH6'
-   | '@OH7'
-   | '@OH8'
-   | '@OH9'
-   | '@OH10'
-   | '@OH11'
-   | '@OH12'
-   | '@OH13'
-   | '@OH14'
-   | '@OH15'
-   | '@OH16'
-   | '@OH17'
-   | '@OH18'
-   | '@OH19'
-   | '@OH20'
-   | '@OH21'
-   | '@OH22'
-   | '@OH23'
-   | '@OH24'
-   | '@OH25'
-   | '@OH26'
-   | '@OH27'
-   | '@OH28'
-   | '@OH29'
-   | '@OH30'
-   ;
-
-hcount
-   : UH
-   | UH DIGIT
-   ;
-
-charge
-   : '-'
-   | '-' DIGIT+
-   | '+'
-   | '+' DIGIT+
-   | '--'
-   | '++'
-   ;
-
-class_
-   : ':' DIGIT+
-   ;
-
-bond
-   : '-'
-   | '='
-   | '#'
-   | '$'
-   | ':'
-   | '/'
-   | '\\'
-   ;
-
-ringbond
-   : bond? DIGIT
-   | bond? '%' DIGIT DIGIT
-   ;
-
-branched_atom
-   : atom ringbond* branch*
-   ;
-
-branch
-   : '(' chain ')'
-   | '(' bond chain ')'
-   | '(' DOT chain ')'
-   ;
-
-chain
-   : branched_atom
-   | chain branched_atom
-   | chain bond branched_atom
-   | chain DOT branched_atom
-   ;
-
-terminator
-   : SPACE TAB
-   | LINEFEED
-   | CARRIAGE_RETURN
-   ;
-
-LA
-   : ('a')
-   ;
-
-LB
-   : ('b')
-   ;
-
-LC
-   : ('c')
-   ;
-
-LD
-   : ('d')
-   ;
-
-LE
-   : ('e')
-   ;
-
-LF
-   : ('f')
-   ;
-
-LG
-   : ('g')
-   ;
-
-LH
-   : ('h')
-   ;
-
-LI
-   : ('i')
-   ;
-
-LJ
-   : ('j')
-   ;
-
-LK
-   : ('k')
-   ;
-
-LL
-   : ('l')
-   ;
-
-LM
-   : ('m')
-   ;
-
-LN
-   : ('n')
-   ;
-
-LO
-   : ('o')
-   ;
-
-LP
-   : ('p')
-   ;
-
-LQ
-   : ('q')
-   ;
-
-LR
-   : ('r')
-   ;
-
-LS
-   : ('s')
-   ;
-
-LT
-   : ('t')
-   ;
-
-LU
-   : ('u')
-   ;
-
-LV
-   : ('v')
-   ;
-
-LW
-   : ('w')
-   ;
+smiles: chain terminator?;
+
+atom: bracket_atom | aliphatic_organic | aromatic_organic | '*';
+
+aliphatic_organic:
+	UB
+	| UC
+	| UN
+	| UO
+	| US
+	| UP
+	| UF
+	| (UC LL)
+	| (UB LR)
+	| UI;
+
+aromatic_organic: LB | LC | LN | LO | LS | LP;
+
+bracket_atom:
+	'[' isotope? symbol chiral? hcount? charge? class_? ']';
+
+isotope: DIGIT+;
+
+symbol: element_symbols | aromatic_symbol | '*';
+
+element_symbols: (UH)
+	| (UH LE)
+	| (UL LI)
+	| (UB LE)
```

Comment:
It would be better to remove useless parentheses:

```suggestion
element_symbols: (UH)
	| UH LE
	| UL LI
	| UB LE
	...
```

Reply:
Fully agree, will add it